src/components/number-input/number-input.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { customElement, html, property, query } from 'lit-element';
+import classnames from 'classnames';
+import settings from 'carbon-components/es/globals/js/settings';
+import WarningFilled16 from '@carbon/icons/lib/warning--filled/16';
+import CaretUp16 from '@carbon/icons/lib/caret--up/16';
+import CaretDown16 from '@carbon/icons/lib/caret--down/16';
+import styles from './number-input.scss';
+import BXInput from '../input/input';
+
+const { prefix } = settings;
+
+@customElement(`${prefix}-number-input`)
+export default class BXNumberInput extends BXInput {
+  @query('input')
+  protected _input!: HTMLInputElement;
+
+  /**
+   * Handles incrementing the value in the input
+   */
+  protected _handleIncrement() {
+    this._input.stepUp();
+  }
+
+  /**
+   * Handles decrementing the value in the input
+   */
+  protected _handleDecrement() {
+    this._input.stepDown();
+  }
+
+  protected _min = '';
+
+  protected _max = '';
+
+  @property({ type: Number, reflect: true })
+  set min(value) {
+    this._min = value;
+  }
+
+  get min() {
+    return this._min.toString();
```

Comment:
Definitly should have commented these ones better ... the `<input type="number">` only respects string values for min/max/step ... Frameworks like Angular end up assigning numeric values and breaking the component (and if you assign to the property directly it would also break the component), the getter/setter just simplifies the value conversion

Reply:
Got it, thanks a lot @cal-smith for explaining this and my apologies for misunderstanding 🙇

Reply:
No worries!